File: Lua/telegram_bot.lua
local telegram = require("telegram")

local TOKEN = "YOUR_TELEGRAM_BOT_TOKEN"
local LOG_PATH = "./logs/default_log.txt"

local bot = telegram(TOKEN)

local function log_message(message)
    local user_info = string.format("%s | User ID: %d | Username: %s | First Name: %s | Last Name: %s | Language: %s",
        os.date('%Y-%m-%d %H:%M:%S', message.date),
        message.from.id,
        message.from.username or 'N/A',
        message.from.first_name or 'N/A',
        message.from.last_name or 'N/A',
        message.from.language_code or 'N/A'
    )

    local chat_info = string.format("Chat Type: %s | Chat ID: %d | Chat Title: %s",
        message.chat.type,
        message.chat.id,
        message.chat.title or 'N/A'
    )

    local log_text = string.format("🔹%s | Chat Info: [%s] | Message Text: %s\n", user_info, chat_info, message.text or 'N/A')

    local file = io.open(LOG_PATH, "a")
    file:write(log_text)
    file:close()
end

bot.run(function(update)
    if update.message then
        local message = update.message
        if message.text == "/start" then
            local reply = string.format("Hello, I'm Zero Bot =>\n👤 User ID: %d\n👥 Chat ID: %d", message.from.id, message.chat.id)
            bot.sendMessage{chat_id=message.chat.id, text=reply}
        end
        log_message(message)
    end
end)

File: Ruby/telegram_bot.rb
require 'telegram/bot'

TOKEN = 'YOUR_TELEGRAM_BOT_TOKEN'

LOG_PATH = "./logs/default_log.txt"

Telegram::Bot::Client.run(TOKEN) do |bot|
  bot.listen do |message|
    case message.text
    when '/start'
      bot.api.send_message(chat_id: message.chat.id, text: "Hello, I'm Zero Bot =>\n👤 User ID: #{message.from.id}\n👥 Chat ID: #{message.chat.id}")
      log_message(message)
    else
      log_message(message)
    end
  end
end

def log_message(message)
  user_info = "#{message.date} | " +
              "User ID: #{message.from.id} | " +
              "Username: #{message.from.username} | " +
              "First Name: #{message.from.first_name} | " +
              "Last Name: #{message.from.last_name} | " +
              "Language: #{message.from.language_code}"

  chat_info = "Chat Type: #{message.chat.type} | " +
              "Chat ID: #{message.chat.id} | " +
              "Chat Title: #{message.chat.title || 'N/A'}"

  log_text = "🔹#{user_info} | Chat Info: [#{chat_info}] | Message Text: #{message.text}\n"
  
  File.open(LOG_PATH, 'a') { |file| file.write(log_text) }
end
